pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -542,3 +554,27 @@ func getAPIResourceConfig(options configapi.MasterConfig) genericapiserver.APIRe
 
 	return resourceConfig
 }
+
+// TODO remove this func in 1.6 when we get rid of the hack above
+func concatenateFiles(prefix, separator string, files ...string) (string, error) {
+	data := []byte{}
+	for _, file := range files {
+		fileBytes, err := ioutil.ReadFile(file)
+		if err != nil {
+			return "", err
+		}
+		data = append(data, fileBytes...)
+		data = append(data, []byte(separator)...)
+	}
+	tmpFile, err := ioutil.TempFile("", prefix)
```

Comment:
This fix is 0600. Good! Maybe add a comment somewhere to calm down somebody reviewing security. The file contains the keys, right?